### PR TITLE
[MI-3423]: Added checks for every post API body

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
https://github.com/mattermost/mattermost-plugin-todo/pull/209

### CODE REVIEW

File: server/plugin.go [hunk 1]
```diff
@@ -163,26 +163,18 @@ func (p *Plugin) checkAuth(handler http.HandlerFunc) http.HandlerFunc {
...
 	userID := r.Header.Get("Mattermost-User-ID")
 
-	var telemetryRequest *telemetryAPIRequest
-	decoder := json.NewDecoder(r.Body)
-	err := decoder.Decode(&telemetryRequest)
+	telemetryRequest, err := GetTelemetryPayloadFromJSON(r.Body)
 	if err != nil {
-		p.API.LogError("Unable to decode JSON err=" + err.Error())
-		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to decode JSON", err)
+		p.API.LogError("Unable to get telemetry payload from json=" + err.Error())
```
Comment (manojmalik20) [inline server/plugin.go]:
```suggestion
		p.API.LogError("Unable to get telemetry payload from JSON err=" + err.Error())
```

File: server/plugin.go [hunk 2]
```diff
@@ -191,33 +183,23 @@ func (p *Plugin) handleTelemetry(w http.ResponseWriter, r *http.Request) {
...
 
-	var addRequest *addAPIRequest
-	decoder := json.NewDecoder(r.Body)
-	err := decoder.Decode(&addRequest)
+	addRequest, err := GetAddIssuePayloadFromJSON(r.Body)
 	if err != nil {
-		p.API.LogError("Unable to decode JSON err=" + err.Error())
-		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to decode JSON", err)
+		p.API.LogError("Unable to get add issue payload from json=" + err.Error())
+		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to get add issue payload from json.", err)
```
Comment (manojmalik20) [inline server/plugin.go]:
```suggestion
		p.API.LogError("Unable to get add issue payload from JSON err=" + err.Error())
		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to get add issue payload from JSON.", err)
```
Fix everywhere.

File: server/plugin.go [hunk 3]
```diff
@@ -502,24 +458,18 @@ func (p *Plugin) handleAccept(w http.ResponseWriter, r *http.Request) {
...
-
 func (p *Plugin) handleComplete(w http.ResponseWriter, r *http.Request) {
 	userID := r.Header.Get("Mattermost-User-ID")
 
-	var completeRequest *completeAPIRequest
-	decoder := json.NewDecoder(r.Body)
-	if err := decoder.Decode(&completeRequest); err != nil {
-		p.API.LogError("Unable to decode JSON err=" + err.Error())
-		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to decode JSON", err)
+	completeRequest, err := GetCompleteIssuePayloadFromJSON(r.Body)
```
Comment (manojmalik20) [inline server/plugin.go:464]: Is the API `CompleteIssue` or `CompleteOAuth`? Check the naming of variables everywhere.

File: server/plugin.go [hunk 4]
```diff
@@ -502,24 +458,18 @@ func (p *Plugin) handleAccept(w http.ResponseWriter, r *http.Request) {
...
+		p.API.LogError("Unable to get complete request payload from json=" + err.Error())
+		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to get complete request from json.", err)
 		return
 	}
 
-	if completeRequest == nil {
-		p.API.LogError("Invalid request body")
-		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to decode JSON", errors.New("invalid request body"))
+	if err := IsCompleteIssuePayloadValid(completeRequest); err != nil {
+		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to validate complete request payload.", err)
```
Comment (manojmalik20) [inline server/plugin.go]:
```suggestion
		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to validate complete issue request payload.", err)
```

File: server/plugin.go [hunk 5]
```diff
@@ -548,25 +498,18 @@ func (p *Plugin) handleComplete(w http.ResponseWriter, r *http.Request) {
...
+		p.API.LogError("Unable to get remove request payload from json=" + err.Error())
+		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to get remove request from json.", err)
 		return
 	}
 
-	if removeRequest == nil {
-		p.API.LogError("Invalid request body")
-		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to decode JSON", errors.New("invalid request body"))
+	if err := IsRemoveIssuePayloadValid(removeRequest); err != nil {
+		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to validate remove request payload.", err)
```
Comment (manojmalik20) [inline server/plugin.go]:
```suggestion
		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to validate remove issue request payload.", err)
```

File: server/plugin.go (hunk 3 above)
Comment (Kshitij-Katiyar) [reply server/plugin.go:464]: complete Issue only.

File: server/plugin.go [hunk 6]
```diff
@@ -503,13 +503,13 @@ func (p *Plugin) handleRemove(w http.ResponseWriter, r *http.Request) {
 
 	removeRequest, err := GetRemoveIssuePayloadFromJSON(r.Body)
 	if err != nil {
-		p.API.LogError("Unable to get remove request payload from json=" + err.Error())
-		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to get remove request from json.", err)
+		p.API.LogError("Unable to get remove request payload from JSON err=" + err.Error())
```
Comment (manojmalik20) [inline server/plugin.go]:
```suggestion
		p.API.LogError("Unable to get remove issue request payload from JSON err=" + err.Error())
```

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**